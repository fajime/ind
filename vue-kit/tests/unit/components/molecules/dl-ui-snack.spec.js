import { mount } from '@vue/test-utils';
import randomIdMixin from '../../../../src/mixins/randomId';
import DlUISnack from '../../../../src/components/molecules/dl-ui-snack/index.vue';
import { nextTick } from 'vue';

describe('dl-ui-snack', () => {
  randomIdMixin.methods.createRandomId = () => 'random-id';
  const wrapper = mount(DlUISnack);

  it('Component Name', () => {
    expect(DlUISnack.name).toBe('dl-ui-snack');
  });

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('methods', () => {
    it('close', () => {
      const wrapperTest = mount(DlUISnack,
        {
          propsData : {
            show : true
          }
        });
      wrapperTest.vm.close();
      expect(wrapperTest.emitted('closed')).toBeTruthy();
      expect(wrapperTest.visible).toBeFalsy();
    });
  });

  describe('watcher', () => {
    it('show - visible and timeout', async () => {
      const stub = jest.fn();
      jest.useFakeTimers();
      const wrapperTest = mount(DlUISnack,
        {
          propsData : {
            show    : false,
            timeout : 1000
          }
        });
      wrapperTest.vm.close = stub;
      await wrapperTest.setProps({ show : true });
      await nextTick();
      jest.runOnlyPendingTimers();
      expect(stub).toHaveBeenCalled();
    });
    it('show - not visible and timeout', async () => {
      const stub = jest.fn();
      jest.useFakeTimers();
      const wrapperTest = mount(DlUISnack,
        {
          propsData : {
            show    : true,
            timeout : 1000
          }
        });
      wrapperTest.vm.close = stub;
      await wrapperTest.setProps({ show : false });
      await nextTick();
      jest.runOnlyPendingTimers();
      expect(stub).toHaveBeenCalled();
    });

    it('show - visible and not timeout', async () => {
      const stub = jest.fn();
      jest.useFakeTimers();
      const wrapperTest = mount(DlUISnack,
        {
          propsData : {
            show    : true,
            timeout : 0
          }
        });
      wrapperTest.vm.close = stub;
      await wrapperTest.setProps({ show : false });
      await nextTick();
      jest.runOnlyPendingTimers();
      expect(stub).toHaveBeenCalled();
    });
  });

});

