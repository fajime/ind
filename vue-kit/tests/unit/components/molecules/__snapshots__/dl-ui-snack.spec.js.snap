// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dl-ui-snack renders correctly 1`] = `
<transition-stub>
  <!--v-if-->
</transition-stub>
`;
